tenseal/cpp/tensors/encrypted_tensor.h

```diff
@@ -269,6 +271,37 @@ class EncryptedTensor {
    protected:
     optional<string> _lazy_buffer;
 
+    void dispatch_jobs(task_t& worker_func, size_t total_tasks) {
+        size_t n_jobs = this->tenseal_context()->dispatcher_size();
+
+        if (n_jobs == 1) {
+            worker_func(0, total_tasks);
+            return;
+        }
+
```

Comment:
As you mentionned, we may need to set n_jobs to be the minimum between n_jobs and total_tasks, otherwise, the PR looks good to me, thanks!